engine/src/flutter/shell/platform/embedder/embedder.cc

```diff
@@ -2416,6 +2416,10 @@ FlutterEngineResult FlutterEngineInitialize(size_t version,
         "Could not infer the Flutter project to run from given arguments.");
   }
 
+  if (args->engine_id != 0) {
```

Comment:
Should this be wrapped in a `SAFE_ACCESS`?

Reply:
It absolutely should. Thanks for catching this!

Reply:
Would it be useful to have a test that launches the engine using a very old embedder API snapshot to verify we don't segfault?

Reply:
I think it'd be useful as regression test in follow-up PR if we actually merged this without the check. But I can still add it if you think it adds value.